input/psbt.go

```diff
@@ -0,0 +1,327 @@
+// construct the derivation path correctly.
+func LocalPaymentBasePoint(fingerprint, coin uint32,
+	desc *keychain.KeyDescriptor) UnknownOption {
+
+	return wrapUnknownOption(
+		PsbtKeyTypeOutputLocalPaymentBasePoint,
+		descBytes(fingerprint, coin, desc),
+	)
+}
+
+// LocalDelayBasePoint returns an UnknownOption for the local delay base point.
+// It requires the wallet's root key fingerprint and coin type to construct the
+// derivation path correctly.
+func LocalDelayBasePoint(fingerprint, coin uint32,
+	desc *keychain.KeyDescriptor) UnknownOption {
+
+	return wrapUnknownOption(
+		PsbtKeyTypeOutputLocalDelayBasePoint,
+		descBytes(fingerprint, coin, desc),
+	)
+}
+
+// LocalHtlcBasePoint returns an UnknownOption for the local HTLC base point.
+// It requires the wallet's root key fingerprint and coin type to construct the
+// derivation path correctly.
+func LocalHtlcBasePoint(fingerprint, coin uint32,
+	desc *keychain.KeyDescriptor) UnknownOption {
+
+	return wrapUnknownOption(
+		PsbtKeyTypeOutputLocalHtlcBasePoint,
+		descBytes(fingerprint, coin, desc),
+	)
+}
+
+// RemoteMultiSigKey returns an UnknownOption for the remote multisig key.
+func RemoteMultiSigKey(key *btcec.PublicKey) UnknownOption {
+	return wrapUnknownOption(
+		PsbtKeyTypeOutputRemoteMultiSigKey,
+		key.SerializeCompressed(),
+	)
+}
+
+// RemoteRevocationBasePoint returns an UnknownOption for the remote revocation
+// base point.
+func RemoteRevocationBasePoint(key *btcec.PublicKey) UnknownOption {
+	return wrapUnknownOption(
+		PsbtKeyTypeOutputRemoteRevocationBasePoint,
+		key.SerializeCompressed(),
+	)
+}
+
+// RemotePaymentBasePoint returns an UnknownOption for the remote payment base
+// point.
+func RemotePaymentBasePoint(key *btcec.PublicKey) UnknownOption {
+	return wrapUnknownOption(
+		PsbtKeyTypeOutputRemotePaymentBasePoint,
+		key.SerializeCompressed(),
+	)
+}
+
+// RemoteDelayBasePoint returns an UnknownOption for the remote delay base
+// point.
+func RemoteDelayBasePoint(key *btcec.PublicKey) UnknownOption {
+	return wrapUnknownOption(
+		PsbtKeyTypeOutputRemoteDelayBasePoint,
+		key.SerializeCompressed(),
+	)
+}
+
+// RemoteHtlcBasePoint returns an UnknownOption for the remote HTLC base point.
+func RemoteHtlcBasePoint(key *btcec.PublicKey) UnknownOption {
+	return wrapUnknownOption(
+		PsbtKeyTypeOutputRemoteHtlcBasePoint,
+		key.SerializeCompressed(),
+	)
+}
+
+// KeyDescriptorFromUnknownValue decodes a value encoded by LocalDesc. It
+// returns, in order:
+// * Root key fingerprint of the wallet
+// * Coin type
+// * A *keychain.KeyDescriptor
+// * A non-nil error on failure or nil error on success
+//
+// The branch is always expected to be 0 and the purpose 1017'.
+func KeyDescriptorFromUnknownValue(value []byte) (
+	uint32, uint32, keychain.KeyDescriptor, error) {
+
+	// Check if we have a bare derivation or a descriptor with a pubkey.
+	if len(value) != 24 && len(value) != 57 {
```

Comment:
nit: could use constants for 24 and 57, for example `baseLen` and `baseLen + btcec.PubKeyBytesLenCompressed` to remove some magic numbers.